File: Python Exams/Basic Exam Collection/Basics Online Exam - 6 and 7 July 2019/03. Travel Agency/3. travelAgency.py
city= input()
package_type= input()
vip = input()
days= int(input())
vip_member = 0

city_list = ["Bansko", "Borovets", "Varna", "Burgas"]
pack_list = ["noEquipment", "withEquipment", "noBreakfast", "withBreakfast"]

if city not in city_list or package_type not in pack_list:
   print('Invalid input!')
   exit()

if days <= 0:
    print('Days must be positive number!')
    exit()

city_dict = {city:{'withEquipment': 100, 'noEquipment': 80, 'withBreakfast':130, 'noBreakfast':100}}
vip_dict = { vip:{'withEquipment': 10, 'noEquipment':5, 'withBreakfast':12, 'noBreakfast':7}}

if vip == 'yes':
    vip_member = vip_dict[vip][package_type]

if days > 7: days -= 1

hostel = city_dict[city][package_type]
hostel = (hostel * (100 - vip_member)) / 100 * days

print(f'The price is {hostel:.2f}lv! Have a nice time!')


'''
Borovets
noEquipment
yes
6
'''

File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 6 and 7 July 2019/03. Travel Agency/3. travelAgency.js
function travelAgency(arg) {
let [city ,type, vip, days] = [arg[0], arg[1], arg[2], +arg[3]]
let vipMember = 0 

 arrCity = ["Bansko", "Borovets", "Varna", "Burgas"]
 arrPack = ["noEquipment",  "withEquipment", "noBreakfast", "withBreakfast"]
 if (!arrCity.includes(arg[0]) ||  !arrPack.includes(arg[1])) {
   console.log('Invalid input!');
   return
 }

 if (+arg[arg.length -1] <= 0) {
   console.log('Days must be positive number!');
   return
 }

  cityObj = {'city':{'withEquipment': 100,'noEquipment': 80, 'withBreakfast':130, 'noBreakfast':100  } }
  vipObj = {'vip':{'withEquipment': 10,'noEquipment':5, 'withBreakfast':12, 'noBreakfast':7}  }
   
  vip === 'yes' ? vipMember = vipObj.vip[type] : null
  days > 7 ? days-=  1 : null

  hostel = cityObj.city[type]  //80
  hostel =  (hostel * (100 - vipMember)) / 100  * days

  console.log(`The price is ${hostel.toFixed(2)}lv! Have a nice time!`);
}

//travelAgency(['Borovets', 'noEquipment', 'yes', '10'])
//travelAgency(['Burgas', 'noBreakfast', 'no', '4' ])